### DISCUSSION TITLE
What qualifies as prematurely in FlushMode.COMMIT

### DISCUSSION THREAD

**Discussion by co-sic** [mikro-orm/mikro-orm#3321] [Q&A] [answered]
The docs state: 

> FlushMode.COMMIT - The EntityManager tries to delay the flush until the current Transaction is committed, although it might flush prematurely too.

So my first questions is, what qualifies as prematurely? 
And as a second question, is there a way to prevent any kind of auto flushes, as long as i do not call flush manually? This would be the preferred behavior for my use case.

**Comment by B4nan** [comment, marked as answer]
Hmm tbh I think I copied this from hibernate docs, and I think it does not apply for us. The `COMMIT` flush mode is previous behaviour (v4 and earlier), where you need to call flush explicitly.
